pkg/cloud-init/cloud-init.go

```diff
@@ -546,6 +550,9 @@ func GenerateLocalData(vmiName string, namespace string, data *CloudInitData) er
 			data.NoCloudMetaData = &NoCloudMetadata{
 				InstanceID: fmt.Sprintf("%s.%s", vmiName, namespace),
 			}
+			if instanceType != "" {
+				data.NoCloudMetaData.InstanceType = instanceType
+			}
```

Comment:
if the `data.NoCloudMetaData.InstanceType` isn't a pointer, I don't think the `if instanceType != ""` if block means much.